chain/chain/src/resharding/manager.rs

```diff
@@ -213,11 +222,24 @@ impl ReshardingManager {
             };
             let mem_changes = trie_changes.mem_trie_changes.as_ref().unwrap();
             let new_state_root = mem_tries.apply_memtrie_changes(block_height, mem_changes);
+            drop(mem_tries);
+
             // TODO(resharding): set all fields of `ChunkExtra`. Consider stronger
             // typing. Clarify where it should happen when `State` and
             // `FlatState` update is implemented.
             let mut child_chunk_extra = ChunkExtra::clone(&chunk_extra);
             *child_chunk_extra.state_root_mut() = new_state_root;
+            // TODO(resharding) - Implement proper congestion info for
+            // resharding. The current implementation is very expensive.
+            if let Some(congestion_info) = child_chunk_extra.congestion_info_mut() {
+                let epoch_id = block.header().epoch_id();
+                let protocol_version = self.epoch_manager.get_epoch_protocol_version(epoch_id)?;
+
+                let trie = tries.get_trie_for_shard(new_shard_uid, new_state_root);
```

Comment:
`bootstrap_congestion_info` requires TrieAccess - can I get it somehow from the memtries above directly?